test/random.go

```diff
@@ -3,51 +3,234 @@ package test
 import (
 	"math"
 	"math/rand"
+	"time"
 )
 
 func RandomBool() bool {
 	return rand.Intn(2) == 0
 }
 
+func RandomDuration() time.Duration {
+	return RandomDurationFromRange(RandomDurationMinimum(), RandomDurationMaximum())
+}
+
+func RandomDurationFromArray(array []time.Duration) time.Duration {
+	if len(array) == 0 {
+		panic("RandomDurationFromArray: array is empty")
+	}
+	return array[rand.Intn(len(array))]
+}
+
+func RandomDurationFromRange(minimum time.Duration, maximum time.Duration) time.Duration {
+	if maximum < minimum {
+		panic("RandomDurationFromRange: maximum is not greater than or equal to minimum")
+	}
+	if minimum < RandomDurationMinimum() {
+		minimum = RandomDurationMinimum()
+	}
+	if maximum > RandomDurationMaximum() {
+		maximum = RandomDurationMaximum()
+	}
+	return minimum + time.Duration(rand.Int63n(int64(maximum-minimum+1)))
+}
+
+func RandomDurationMaximum() time.Duration {
+	return 10 * 365 * 24 * time.Hour
```

Comment:
they are all constants??

Reply:
Is is the maximum random duration. So the maximum duration that the random duration generators should generate. I name it this way for consistency and scoping reasons. That is everything starts with the prefix + "Maximum", "Minimum", etc...

Reply:
hmmm maybe in context :) I guess I keep thinking it needs a comment ... and then if it needs a comment it need renaming. anyway not blocking